roles/cli/cachetool/tasks/main.yml

```diff
@@ -13,15 +13,66 @@
   when:
     - deploy_operation == 'deploy'
 
-- name: Download latest cachetool installer.
-  get_url:
-    url: "http://gordalina.github.io/cachetool/downloads/cachetool.phar"
-    dest: "{{ cachetool_bin }}"
-    mode: 0755
+- name: Download cachetool depending on latest php version installed. # If not specified manually, according to https://github.com/gordalina/cachetool#compatibility
+  block:
+    - name: Get latest php installed
+      ansible.builtin.shell:
+        cmd: "ls /etc/php/ | tail -1 | sed -e 's/\\.//g'"
+      register: _php_version_str
+
+    - name: Download latest cachetool installer if PHP is 8.1 or newer.
+      get_url:
+        url: "http://gordalina.github.io/cachetool/downloads/cachetool.phar"
+        dest: "{{ cachetool_bin }}"
+        mode: 0755
+      when:
+        - _php_version_str.stdout | int >= 81
+
+    - name: Download cachetool version 8.5.0 installer if PHP is 8.0.
+      get_url:
+        url: "http://gordalina.github.io/cachetool/downloads/cachetool-8.5.0.phar"
+        dest: "{{ cachetool_bin }}"
+        mode: 0755
+      when:
+        - _php_version_str.stdout | int == 80
+
+    - name: Download cachetool version 7.1.0 installer if PHP is 7.3 or newer.
+      get_url:
+        url: "http://gordalina.github.io/cachetool/downloads/cachetool-7.1.0.phar"
+        dest: "{{ cachetool_bin }}"
+        mode: 0755
+      when:
+        - _php_version_str.stdout | int >= 73
+        - _php_version_str.stdout | int < 80
+
+    - name: Download cachetool version 5.1.3 installer if PHP is 7.2.
+      get_url:
+        url: "http://gordalina.github.io/cachetool/downloads/cachetool-5.1.3.phar"
+        dest: "{{ cachetool_bin }}"
+        mode: 0755
+      when:
+        - _php_version_str.stdout | int == 72
+
+    - name: Download cachetool version 4.1.1 installer if PHP is 7.1.
+      get_url:
+        url: "http://gordalina.github.io/cachetool/downloads/cachetool-4.1.1.phar"
+        dest: "{{ cachetool_bin }}"
+        mode: 0755
+      when:
+        - _php_version_str.stdout | int == 71
+
+    - name: Download cachetool version 3.2.2 installer if PHP version is too old.
+      get_url:
+        url: "http://gordalina.github.io/cachetool/downloads/cachetool-3.2.2.phar"
+        dest: "{{ cachetool_bin }}"
+        mode: 0755
+      when:
+        - _php_version_str.stdout | int < 71
+
   when:
     - deploy_operation == 'deploy'
     - not cachetool_global.stat.exists
-    - cachetool.version == 'latest'
+    - cachetool.version is not defined
```

Comment:
This probably won't work - `cachetool.version` *is* defined because it is set to an empty string in defaults. You'll need `cachetool.version | length == 0` similar to the task below.